### PR TITLE
22 create auth app

### PR DESCRIPTION
**Pull request by hrytskivr**
## TRELLO

* [Main Trello ticket](https://trello.com/c/SJ0xwKoT/22-create-auth-app)

## Code reviewers

- [ ] @Ashiato
- [ ] @black-art
- [ ] @KarnaushenkoVO 

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## Summary of issue

"registration" app pushed

## Summary of change

"auth_reg" app have been redone, now it's "registration" app
no more class-based views, using function ones instead
pylint 9.66 and above

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: myTrip/registration/views.py [hunk 1]
```diff
@@ -1,3 +1,73 @@
...
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.http import HttpResponse, JsonResponse
+from django.shortcuts import redirect, render
+
+from .models import CustomUser
+
+
+def auth(request):
+    """Login page"""
```
Comment (black-art) [inline myTrip/registration/views.py]: Add dots in comments in this file.

File: myTrip/registration/views.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                CustomUser.objects.get(email=email)
+                json['error'] = "This email is already registered."
+            except CustomUser.DoesNotExist:
+                user = CustomUser()
+                user.email, user.first_name, user.last_name = email, data.get(
+                    'firstName'), data.get('lastName')
+                user.set_password(data.get('password'))
+
+                user.save()
+                json['success'] = True
```
Comment (KarnaushenkoVO) [inline myTrip/registration/views.py]: You should work with db only in models.py

File: myTrip/registration/models.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+     :argument email: varchar - user's emailAdress
+     """
+
+    first_name = models.CharField(max_length=254, blank=True)
+    last_name = models.CharField(max_length=254, blank=True)
+    email = models.EmailField(blank=False, unique=True)
+
+    USERNAME_FIELD = 'email'
+
+    def create_user(self, email, password):
```
Comment (lhalam) [inline myTrip/registration/models.py]: use static method
don't use class name in methods

File: myTrip/registration/models.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        """
+        Creates and saves a User with the given email and password.
```
Comment (lhalam) [inline myTrip/registration/models.py]: for example:
```
"""
Get Like with given trip id
 Args:
    trip_id (int): trip id.
Returns:
    QuerySet<Like>: QuerySet of Like.
"""
```

File: myTrip/registration/models.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            user = CustomUser.objects.get(id=user_id)
+        except CustomUser.DoesNotExist:
+            return None
+
+        return user
+
+
+    def to_dict(self):
+        """
+        Convert model object to dictionary.
```
Comment (lhalam) [inline myTrip/registration/models.py]: for example:
```
"""Convert model object to dictionary.
        Return:
            dict:
                {
                    'id': id,
                    'user_id': name,
                    ...
                }
        """
```
__

File: myTrip/registration/views.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+from django.contrib import auth as authentication
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.http import HttpResponse, JsonResponse
+from django.shortcuts import redirect, render
+
+from .models import CustomUser
+
+
+def auth(request):
```
Comment (lhalam) [inline myTrip/registration/views.py]: delete this func

File: myTrip/registration/views.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    """Login page"""
+
+    return render(request, '')
+
+def logout(request):
+    """Logout method"""
+
+    authentication.logout(request)
+    return redirect('')
```
Comment (lhalam) [inline myTrip/registration/views.py]: return httpstatus

File: myTrip/registration/views.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+    if request.method == 'POST':
+        data = loads(request.body.decode('utf-8'))
+        email = data.get('email').strip().lower()
+        password = data.get('password').strip()
+
+        user = authentication.authenticate(username=email, password=password)
+        if user:
+            authentication.login(request, user)
+            return JsonResponse({'success': True, 'message': '/'})
```
Comment (lhalam) [inline myTrip/registration/views.py]: HttpResponse(status=200)

File: myTrip/registration/views.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+        return HttpResponse('Email and/or password invalid', status=403)
+
+    return HttpResponse(status=400)
+
+def register(request):
+    """Registration method for CustomUser registration"""
+
+    if request.method == 'POST':
+        json = {
```
Comment (lhalam) [inline myTrip/registration/views.py]: delete this obj

File: myTrip/registration/views.py [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        data = loads(request.body.decode('utf-8'))
+        email = data.get('email')
+
+        try:
+            validate_email(email)
+            try:
+                CustomUser.objects.get(email=email)
+                json['error'] = "This email is already registered."
```
Comment (lhalam) [inline myTrip/registration/views.py]: return HttpResponse(status=400)

**Review by black-art — APPROVED** [review]

**Pull request merged**